linchpin/api/__init__.py

```diff
@@ -70,16 +113,31 @@ def run_playbook(self, pinfile, targets='all', playbook='provision'):
                                     self.ctx.cfgs['evars']['layouts_folder'],
                                     pf[target]["layout"]))
 
-
-#                def invoke_linchpin(ctx, lp_path, self.ctx.evars, playbook='provision', console=True):
+                # set the current target data 
+                self.current_target_data = pf[target]
+                self.current_target_data["extra_vars"] = self.ctx.evars
+                # set the state to preup/predown based on playbook
+                # note : changing the state triggers the hooks
+                if playbook == "provision":
```

Comment:
Hardcoding this here means that you can't predict what the playbook is going to be. In fact, in upcoming changes the playbook is no longer called 'provision', but rather 'up'. Hardcoding this in means it has to be changed in every place it's updated. Instead, consider getting the list of states 'up', 'destroy', 'down', etc., from configurations and assigning them to a variable.